pkg/config/s3.go

```diff
@@ -2,9 +2,10 @@ package config
 
 // S3Config specifies the properties required to use S3 as the storage backend
 type S3Config struct {
-	Region string `validate:"required" envconfig:"AWS_REGION"`
-	Key    string `envconfig:"AWS_ACCESS_KEY_ID"`
-	Secret string `envconfig:"AWS_SECRET_ACCESS_KEY"`
-	Token  string `envconfig:"AWS_SESSION_TOKEN"`
-	Bucket string `validate:"required" envconfig:"ATHENS_S3_BUCKET_NAME"`
+	Region                string `validate:"required" envconfig:"AWS_REGION"`
+	Key                   string `envconfig:"AWS_ACCESS_KEY_ID"`
+	Secret                string `envconfig:"AWS_SECRET_ACCESS_KEY"`
+	Token                 string `envconfig:"AWS_SESSION_TOKEN"`
+	Bucket                string `validate:"required" envconfig:"ATHENS_S3_BUCKET_NAME"`
+	UseAmbientCredentials bool   `envconfig:"AWS_USE_AMBIENT_CREDENTIALS"`
```

Comment:
Can we use a better name?
Also, can we add it to config.dev.toml so that users know that this one exists?
Let us make sure that we add documentation for that. Right from the docs, I think this was the best  description, I got: 
```
session Provides initial default configuration, and load
      configuration from external sources such as environment and shared
      credentials file.
```
but feel free to improvize

Reply:
Yes, I can update the naming. What would you call it?

Reply:
`AWS_USE_DEFAULT_CONFIGURATION`?